controllers/serviceaccount_controller.go

```diff
@@ -257,8 +259,8 @@ func (r *ServiceAccountReconciler) ensureProviderServiceAccounts(ctx context.Con
 		log := log.WithValues("EnsureProviderServiceAccount", klog.KRef(pSvcAccount.Namespace, pSvcAccount.Name))
 		ctx := ctrl.LoggerInto(ctx, log)
 
-		if guestClusterCtx.Cluster != nil && annotations.IsPaused(guestClusterCtx.Cluster, &(pSvcAccounts[i])) {
-			log.V(4).Info("Skipping ensure ProviderServiceAccount as ProviderServiceAccount is paused or belongs to a cluster that is paused ")
+		if annotations.HasPaused(&(pSvcAccounts[i])) {
```

Comment:
.Cluster was always nil because before this PR we didn't set it in the cluster context

I think it's fine to not check it here as we already check if the cluster is paused above